app/src/org/commcare/android/adapters/IncompleteFormListAdapter.java

```diff
@@ -306,25 +334,41 @@ private void filterValues() {
 
         current.clear();
 
-        if (query == null || query.equals("")) {
+        if ("".equals(query)) {
```

Comment:
you've changed the truth value here, just double checking that that was intentional